api/src/main/java/org/apache/iceberg/transforms/Transforms.java

```diff
@@ -40,6 +41,7 @@ private Transforms() {
   }
 
   private static final Pattern HAS_WIDTH = Pattern.compile("(\\w+)\\[(\\d+)\\]");
+  private static final Pattern HAS_TIME_OFFSET = Pattern.compile("(\\w+)\\[(\\-?\\d+)\\]");
```

Comment:
The pattern should require either `-` or `+`, and should also require 2-digit values for hour and minute. Using hour/minute instead of seconds makes this much easier to read and work with for users.